### ISSUE TITLE
🐛 `require is not defined`

### ISSUE THREAD

**Issue by benbonnet** [parcel-bundler/parcel#40]
Is that just me ? [the simplest example](https://parceljs.org/getting_started.html) does not even work; breaking on the very first variable in ./build/index.js :
```
"use strict";
require = function (_require) { etc…
```
Or transforms/babelization/etc…  (not shown in the example) are to be added first ?

**Comment by handoing** [comment]
I have the same problem.

**Comment by deesx** [comment]
Just tried out this one : https://github.com/davidsonsns/parcel-react-app

Was on node6, upgrading to node8 let me run this example without issues.

node8 is obviously required; so is a minimal .babelrc setup

**Comment by labnol** [comment]
Getting the same issue on Node 6

**Comment by thejameskyle** [comment]
We're relying on non-strict mode. Some tool (most likely Babel is adding in the `"use strict"`.

We should probably update the [prelude.js](https://github.com/parcel-bundler/parcel/blob/ad5135adfda84e7bf49a9aee477deae7d83a3588/src/builtins/prelude.js) file to find the correct global to attach to.

@ljharb What's the best way to get `global` (whether it be `window/self/global/this`) today?

**Comment by bigtunacan** [comment]
Just chiming in; I was getting the a different error #31, but after that issue was fixed I'm now getting this error with node6.

**Comment by ljharb** [comment]
In strict or sloppy mode, you can always use `Function(‘return this’)()` in every engine except CSP-enabled browsers (however, for those, you can use `window || self`.

**Comment by handoing** [comment]
I upgraded node v9.2.0 and solved it.

**Comment by marcioj** [comment]
I was looking into the issue but I think even fixing the use strict problem we will have problems with babel transforming `require = (function (modules, cache, entry) { ... })` to `require = function (modules, cache, entry) { ... };` when the project is built. 
So in node 7 and further, when the script is compiled by parcel and the browser loads, we have correctly `require = (function (modules, cache, entry) { ... })([modules])` which is a iife. Because it uses the files under src/* which are not compiled it uses the original file. But in node 6 and before it generates `require = function (modules, cache, entry) { ... };([modules])`. Which makes the page render nothing.

Not sure how to fix that. Maybe we could make babel ignore this specific file and just copy it to lib?

Also is it a bad idea to just not support node 6? I think people using parcel are mostly testing it or creating new stuff, in this case they can at least use a more up to date node version.

**Comment by bigtunacan** [comment]
I just ended up upgrading to Node 8x series since it is on LTS

**Comment by ollieSk8** [comment]
just upgrade the node version to v8.9.3,can fix this problem.

**Comment by xswei** [comment]
my node version is v6,  this issue was be fixed after upgrade to v8

**Comment by wqcstrong** [comment]
@xswei me,too! But do you know why?

**Comment by phobal** [comment]
i try it, since v7.0.0 to v7.9.0, it cannot run, but when I use v8.0.0, it done !, so, I guess parcel need Node version is greater than or equal to v8.0.0

**Comment by wqcstrong** [comment]
@phobal But what results to the problem, I would like to konw the explaniation. It's all konwn that ```require``` is supported at the node @v6.x.x

**Comment by phobal** [comment]
@wqcstrong right. node 6.x was supported, however, it run in node V8 environment, in browser cannot support `require`

**Comment by xswei** [comment]
@wqcstrong  @phobal  老铁  parcel只支持8以上版本，具体原因知道吗？

**Comment by guilhermeasg** [comment]
Also bumped into this on Node 6.11. The first few lines of the generated file follow. Note the implicit global assignment in `require = function(...)`, which is the reason the error is thrown in strict mode. Manually deleting the `use strict` line, or using `var require = function...` prevents the error from being thrown, but the generated file actually never calls the `require` function.

```js
"use strict";

// modules are defined as an array
// [ module function, map of requires ]
//
// map of requires is short require name -> numeric require
//
// anything defined in a previous bundle is accessed via the
// orig method which is the require for previous bundles

require = function (_require) {
```

**Comment by wqcstrong** [comment]
@xswei 上面的大佬大概说了导致报错的原因了 @guilhermeasg ：1、删除严格模式；2、```var```声明变量

**Comment by wqcstrong** [comment]
@guilhermeasg Hey buddy,I have tried to find the semicolon you said in the complied file, but I did not find it  after reading what you said.

**Comment by guilhermeasg** [comment]
@wqcstrong thanks for following up. I uploaded the bundled file to [a gist](https://gist.github.com/guilhermeasg/f5e673b019551d0c0b18e76656b9f943) for direct examination. This file has been generated, as in the example in the main page, simply by running `parcel index.html` on the provided file. 

Note that line 11 is an implicit global definition, which is forbidden in strict mode (L1), hence the error. Furthermore, take a look at L80:

```js
});({7:[function(require,module,exports) {
```

The expression after the semicolon is supposed to be the argument to the previously defined function (making it an IIFE), but the semicolon breaks it into an object declaration with no side effects.

**Comment by purtuga** [comment]
Looking forward to this fix pending (#253).

To the project maintainers:
Having recently heard about Parcel and it's zero configuration feature, I was eager to try it out on a new project, but quickly found that the "Getting Started" example just fails, which needless to say is not a good reflection upon this library.  My suggestion (until this issue is fixed) would be for you to list the current limitation (requires nodeJS v > 6) on your README file and website. 

Appreciate the work that has gone into this library and for the introduction of a zero config bundler. I will revisit this again in the future...

**Comment by perymimon** [comment]
what wrong to write 
`var require = function (_require) {`
instead of just
`require = function (_require) {` ?

It make it global just fine

**Comment by guilhermeasg** [comment]
@perymimon see [my comment](https://github.com/parcel-bundler/parcel/issues/40#issuecomment-351018706) for the reason

**Comment by mike-rotate** [comment]
Just as an update:

I'm getting this issue with parcel bundler version 1.4.1 & node version v9.4.0.

Both fresh installs today.

**Comment by davidnagli** [comment]
@mike-rotate Can you try it again using the new Parcel version?

**Comment by V1cHu** [comment]
Still getting the same issue. 

Node version : 9.5.0
Parcel version : 1.5.1

Just have a button in my html page which on clicking calls the clickHandler function in the js file. Get an error saying 'ReferenceError: clickHandler is not defined'. I could find the function in the built js file under dist.

Note: have configured to use babel. But still..  😞

**Comment by devongovett** [comment]
Node 6 support was just merged in #789. Should be released in 1.6.0 this week!
